### PR TITLE
[2/3] Create Tag with visualID/ adapt unit test tags

### PR DESCRIPTION
**Pull request by mahdibenromdhane**

### CODE REVIEW

**Comment by jerome-benoit** [comment]
@mahdibenromdhane: Your plugins on vscode that should lint and fix the code are not properly configured. Please fix your setup and run `npm run eslint:fix` on your work branches.

File: src/server/rest/v1/service/TagService.ts [hunk 1]
```diff
@@ -143,6 +143,18 @@ export default class TagService {
         action: action
       });
     }
+    // Check Tag
+    tag = await TagStorage.getTagByVisualID(req.user.tenantID, filteredRequest.visualID);
+    if (tag) {
+      throw new AppError({
+        source: Constants.CENTRAL_SERVER,
+        errorCode: HTTPError.TAG_VISUAL_ID_ALREADY_EXIST_ERROR,
+        message: `Tag with ID '${filteredRequest.id}' already exists`,
```
Comment (jerome-benoit) [inline src/server/rest/v1/service/TagService.ts:152]: The error message is misleading: you look for for duplicate visualID records and the message is refereeing to the tag id.

File: src/server/rest/v1/service/TagService.ts [hunk 2]
```diff
@@ -461,7 +474,7 @@ export default class TagService {
             if ((tagsToBeImported.length % Constants.IMPORT_BATCH_INSERT_SIZE) === 0) {
               await TagService.insertTags(req.user.tenantID, req.user, action, tagsToBeImported, result);
             }
-          // eslint-disable-next-line @typescript-eslint/no-misused-promises
+            // eslint-disable-next-line @typescript-eslint/no-misused-promises
```
Comment (jerome-benoit) [inline src/server/rest/v1/service/TagService.ts:477]: Previous indentation was correct.

File: src/server/rest/v1/service/TagService.ts [hunk 3]
```diff
@@ -478,8 +491,8 @@ export default class TagService {
               res.writeHead(HTTPError.INVALID_FILE_FORMAT);
               res.end();
             }
-          // Completed
-          // eslint-disable-next-line @typescript-eslint/no-misused-promises
+            // Completed
+            // eslint-disable-next-line @typescript-eslint/no-misused-promises
```
Comment (jerome-benoit) [inline src/server/rest/v1/service/TagService.ts:495]: Previous indentation was correct.

**Comment by mahdibenromdhane** [comment]
> @mahdibenromdhane: Your plugins on vscode that should lint and fix the code are not properly configured. Please fix your setup and run `npm run eslint:fix` on your work branches.

Done.

File: src/storage/mongodb/TagStorage.ts [hunk 4]
```diff
@@ -284,7 +295,7 @@ export default class TagStorage {
   public static async getTags(tenantID: string,
       params: {
         issuer?: boolean; tagIDs?: string[]; userIDs?: string[]; dateFrom?: Date; dateTo?: Date;
-        withUser?: boolean; withUsersOnly?: boolean; withNbrTransactions?: boolean; search?: string, defaultTag?: boolean, active?: boolean
+        withUser?: boolean; withUsersOnly?: boolean; withNbrTransactions?: boolean; search?: string, defaultTag?: boolean, active?: boolean, visualID?: string[]
```
Comment (jerome-benoit) [inline src/storage/mongodb/TagStorage.ts:298]: The added argument naming and ordering is inconsistent: 
Please create an helper method `getTagByVisualIDs`, rename `visualID?` to `visualIDs?` and put `visualIDs?` param after `userIDs?`

And reuse `getTagByVisualIDs` in `getTagByVisualID`

**Review by jerome-benoit — CHANGES_REQUESTED** [review]

**Review by jerome-benoit — APPROVED** [review]

**Pull request merged**